src/test/java/io/cdap/plugin/aws/sqs/sink/SqsSinkConfigTest.java

```diff
@@ -0,0 +1,313 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.aws.sqs.sink;
+
+import com.amazonaws.regions.Regions;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.aws.sqs.sink.util.SinkMessageFormat;
+import io.cdap.plugin.aws.sqs.util.SqsAuthMethod;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+import static io.cdap.plugin.aws.sqs.sink.SqsSinkConfigHelper.TEST_ACCESS_ID;
+import static io.cdap.plugin.aws.sqs.sink.SqsSinkConfigHelper.TEST_ACCESS_KEY;
+import static io.cdap.plugin.aws.sqs.sink.SqsSinkConfigHelper.TEST_QUEUE_NAME;
+import static io.cdap.plugin.aws.sqs.sink.SqsSinkConfigHelper.TEST_REGION;
+import static io.cdap.plugin.aws.sqs.sink.SqsSinkConfigHelper.TEST_SQS_ENDPOINT;
+import static io.cdap.plugin.aws.sqs.sink.util.SqsSinkConstants.PROPERTY_DELAY_SECONDS;
+import static io.cdap.plugin.aws.sqs.sink.util.SqsSinkConstants.PROPERTY_MESSAGE_FORMAT;
+import static io.cdap.plugin.aws.sqs.util.SqsConstants.PROPERTY_ACCESS_ID;
+import static io.cdap.plugin.aws.sqs.util.SqsConstants.PROPERTY_ACCESS_KEY;
+import static io.cdap.plugin.aws.sqs.util.SqsConstants.PROPERTY_AUTH_METHOD;
+import static io.cdap.plugin.aws.sqs.util.SqsConstants.PROPERTY_QUEUE_NAME;
+import static io.cdap.plugin.aws.sqs.util.SqsConstants.PROPERTY_REGION;
+
+/**
+ * Tests for {@link SqsSinkConfig}.
+ */
+public class SqsSinkConfigTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testAuthMethodCredentials() {
+    SqsAuthMethod authMethod = SqsAuthMethod.ACCESS_CREDENTIALS;
+    SqsSinkConfig config = SqsSinkConfigHelper.newConfigBuilder()
+      .setAuthenticationMethod("Access Credentials")
+      .build();
+
+    MockFailureCollector collector = new MockFailureCollector();
+    Assert.assertEquals(authMethod, config.getAuthenticationMethod(collector));
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testAuthMethodIAM() {
+    SqsSinkConfig config = SqsSinkConfigHelper.newConfigBuilder()
+      .setAuthenticationMethod("IAM")
+      .build();
+
+    try {
+      MockFailureCollector collector = new MockFailureCollector();
+      config.getAuthenticationMethod(collector);
```

Comment:
Instead of throwing the error in `getAuthenticationMethod` and catching it, you can check the contents of `MockFailureCollector` to see if the error was generated properly. This comment applies to all of the validate methods

Reply:
Done